cpp/src/traversal/tsp_utils.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+/* CPU side validation */
+#define cpudist(a, b)                                                                          \
+  (sqrtf((pos[a] - pos[b]) * (pos[a] - pos[b]) + (pos[a + nodes_ + 1] - pos[b + nodes_ + 1]) * \
+                                                   (pos[a + nodes_ + 1] - pos[b + nodes_ + 1])))
+
+// Round all distances to the nearest integer, which is good enough for realistic distances in ft
+#define tilesize 128
+#define kswaps 4
+#define dist(a, b) \
```

Comment:
Same comment as above: please consider using an inlined (template) function instead of macro.